apps/sports-helsinki/src/domain/venue/useVenueCards.tsx

```diff
@@ -0,0 +1,45 @@
+import type { Venue } from 'events-helsinki-components';
+import { useLocale } from 'events-helsinki-components';
+import React from 'react';
+import {
+  Card,
+  getLocationCardProps,
+  useConfig,
+} from 'react-helsinki-headless-cms';
+import { ROUTES } from '../../constants';
+import routerHelper from '../../domain/app/routerHelper';
+
+type useVenueCardsProps = {
+  venues?: Venue[] | null;
+  returnPath?: string;
+};
+
+function useVenueCards({ venues, returnPath }: useVenueCardsProps) {
```

Comment:
Maybe there's a good reason, but I'm just wondering why is this a hook and not just react component (ie. `VenueCardList.tsx` which would return a list of Card components.
It could return
```
<>
 {{ venues?.map((venue, i) => { ... <Card .../>})}}
</>
```
I would see that hook would rather implement some business logic instead of returning components.

Reply:
The main reason is the type of the HCRC-lib's cards-property:
```
cards: React.ReactElement<({ id, ariaLabel, className, imageUrl, imageLabel, title, subTitle, text, customContent, hasLink, linkArrowLabel, url, withBorder, withShadow, direction, clampText, openLinkInNewTab, style }: CardProps) => JSX.Element, string | React.JSXElementConstructor<any>>[]
```

Would it be ok to not use a hook, but get the cards like this `const cards = VenueCardList({ venues, returnPath: router.asPath });` I find it out even more confusing.

Using the collection like this is not valid for HCRC
```
            <Collection
              type={type}
              title={t('venue:similarVenues.title')}
              cards={<VenueCardList venues={venues} returnPath={router.asPath}}
              loading={loading}
            />
```